### PR TITLE
DB Deployment problems on travis and local development

### PR DESCRIPTION
**Pull request by roy-zahor**
HOTFIX to deployment problem found after merging #338

**The cause:** 
create_db.sql wasn't tolerant to running twice in a row, or in some cases while running on a system with Spark DB already set up.

**The solution:** 
Fixing travis scripts to run mysql DB creation and npm test only once.
Updating documentation.

**Bonus feature:** 
Added **npm run-script createdb** as a wrapper for **mysql -u root < migrations/create_db.sql**

### CODE REVIEW

**Comment by OriHoch** [comment]
indeed create_db is only meant to run once

and to the best of my knowledge that's exactly what happens

teavis has new environment on every run

I don't understand what's the problem you are trying to fix

travis failures were due to migrations files which were deleted

this can be fixes by either recreating test environment db from scratch
or (better) manually sync the db state to the migrations state (as you will need to do the same for production)

**Comment by OriHoch** [comment]
also, I see you moved deploy to run before the tests and as part of travis 'script' section, this is a bad practice in my opinion, we should deploy only if tests pass
and also, it should be in the appripriate travis section for deploy... (there are some differences in how the sections behave, mostly regarding how errors are handles)

File: migrations/create_db.sql [hunk 1]
```diff
@@ -1,5 +1,10 @@
 DROP DATABASE IF EXISTS spark;
 
+# Implicitly create the user and then drop the user.
+GRANT USAGE ON *.* TO 'spark'@'localhost' IDENTIFIED BY 'password';
+DROP USER 'spark'@'localhost';
+flush privileges;
```
Comment (OriHoch) [inline migrations/create_db.sql:6]: why?

**Comment by OriHoch** [comment]
I added a note to Codeship documentation to warn that it's not actually used, all build / test / deploy processes are on travis
https://github.com/Midburn/Spark/blob/master/.codeship/README.md

**Comment by roy-zahor** [comment]
@OriHoch The fact that knex found some missing migrations is because the DB stayed "dirty" from the previous travis run. Otherwise knex should not be aware that migrations are missing (knex simply scans the migrations folder to find migrations).

File: migrations/create_db.sql (hunk 1 above)
Comment (roy-zahor) [reply migrations/create_db.sql:6]: This 3 lines are the same as "drop user if exists", which does not exists in mysql (up until last version, and at this point of time I can't rely on it)

**Comment by roy-zahor** [comment]
@OriHoch deploy.sh was already in the "scripts" section, I just moved it to the beginning, but since It doesn't really matter, I'm moving it back to the original location.

**Comment by OriHoch** [comment]
@roy-zahor there are 2 steps in travis:
* step 1 - run the tests
  * this is almost all the travis commands you see, like creating the DB, bower install, etc..
  * all these commands run on a fresh travis environment which is created from scratch each time
* step 2 - deploy
  * this step runs only when merging to master branch
  * it does SSH to test server and deploys there
  * this stage is implemented in the deploy.sh script

the problem after your merge are in the 2nd step - on the test environment server. You don't see these errors here on the pull requests because we don't deploy pull requests, only merges to master.

File: migrations/create_db.sql (hunk 1 above)
Comment (OriHoch) [reply migrations/create_db.sql:6]: yes, but why do we need to do that? I don't strongly object to it, just that I rather let whomever is calling this script to deal with users permissions
(e.g. on travis the user is `root` and `password` is empty - no need to add a different user)

Comment (roy-zahor) [reply migrations/create_db.sql:6]: Good point, we should discuss it. My code change doesn't change the behavior of user creation, it simply address an issue that might rise when running the same script more than once.
In the meantime, I want to commit to resolve issues on master.

**Pull request merged**